nullaway/src/main/java/com/uber/nullaway/dataflow/ArrayIndexElement.java

```diff
@@ -0,0 +1,61 @@
+package com.uber.nullaway.dataflow;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+import javax.lang.model.element.Element;
+
+public class ArrayIndexElement implements AccessPathElement {
+  private final Element javaElement;
+  @Nullable private final Integer constantIndex;
+  @Nullable private final Element variableElement;
```

Comment:
Rather than having two `@Nullable` fields here to represent the cases, I'd prefer a single field of type `Object` named `index` for simplicity and efficiency.  The API can still be type safe (i.e., you can only pass in an `Integer` or an `Element`); the fact that the same field is used internally would be a private implementation detail.  You can document the expectations in a comment.